katsdptelstate/rdb_writer.py

```diff
@@ -16,145 +16,146 @@
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Exit RDB writer context, closing off file and deleting it if empty."""
+        try:
+            self._rdb_context.__exit__(exc_type, exc_value, traceback)
+        finally:
+            if not self.keys_written:
+                logger.error("No valid keys found - removing empty file")
+                os.remove(self.filename)
+        return False
 
-        Subsequent bytes represent the name of the key, which consists of a length
-        encoding followed by the actual byte representation of the string. For this simple
-        writer we only provide two length encoding schemas:
-             String length up to 63 - single byte, 6 LSB encode number directly
-             String length from 64 to 16383 - two bytes, 6 LSB of byte 1 + 8 from byte 2 encode number
+    def save(self, client, keys=None):
+        """Save a specified subset of keys from a Redis DB to the RDB dump file.
 
-        Thereafter the value, encoding according to its appropriate schema is appended. As a shortcut,
-        the Redis DUMP command is used to generate the encoded value string.
+        This is a very limited RDB dump utility. Either a compatible redis client
+        or a :class:`katsdptelstate.Backend` must be provided. Missing or broken
+        keys are skipped and logged without raising an exception.
 
-        Note: Redis provides a mechanism for optional key expiry, which we ignore here.
+        Parameters
+        ----------
+        client : :class:`~redis.StrictRedis` or :class:`~katsdptelstate.telescope_state.Backend`
+            A Redis-compatible client instance. Must support keys() and dump().
+        keys : sequence of str or bytes, optional
+            The keys to extract from Redis and include in the dump.
+            Keys that don't exist will not raise an Exception, only a log message.
+            None (default) includes all keys.
         """
-        if not key_dump:
-            raise KeyError('Key {} not found in Redis'.format(key))
-        if not isinstance(key, bytes):
-            key = key.encode('utf-8')
-        try:
-            key_len = encode_len(len(key))
-        except ValueError as e:
-            raise ValueError('Failed to encode key length: {}'.format(e))
-        key_type = key_dump[:1]
-        encoded_val = key_dump[1:-10]
-         # The DUMPed value includes a leading type descriptor, the encoded value itself (including length specifier),
-         # a trailing version specifier (2 bytes) and finally an 8 byte checksum.
-         # The version specified and checksum are discarded.
-        return key_type + key_len + key + encoded_val
+        if keys is None:
+            logger.warning("No keys specified - dumping entire database")
+            keys = client.keys(b'*')
+        for key in keys:
+            key = _ensure_binary(key)
+            key_dump = client.dump(key)
```

Comment:
As above, `key_dump` is a misleading name since this contains the value and not the key.

Reply:
Already fixed.